sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java

```diff
@@ -0,0 +1,274 @@
+        .jersey()
+        .register(new DelegatingOpenApiResource(instanceId).openApiConfiguration(oasConfig));
+
+    // Allow CORS to access (via wildcard) from Swagger UI/editor
+    String basePath = determineBasePath(configuration);
+
+    String filterBasePath = basePath.endsWith("/") ? basePath : basePath + "/"; // NOSONAR
+    FilterRegistration.Dynamic filter =
+        environment.servlets().addFilter("CORS OpenAPI", CrossOriginFilter.class);
+    filter.addMappingForUrlPatterns(
+        EnumSet.allOf(DispatcherType.class),
+        true,
+        filterBasePath + "openapi.yaml",
+        filterBasePath + "openapi.json");
+    filter.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
+    filter.setInitParameter(CrossOriginFilter.ALLOW_CREDENTIALS_PARAM, Boolean.TRUE.toString());
+    filter.setInitParameter(CrossOriginFilter.CHAIN_PREFLIGHT_PARAM, Boolean.FALSE.toString());
+
+    LOG.info(
+        "Initialized OpenAPI with base path '{}' and resource packages: '{}'",
+        basePath,
+        resourcePackages);
+  }
+
+  private String determineBasePath(Configuration configuration) {
+    ServerFactory serverFactory = configuration.getServerFactory();
+
+    if (serverFactory instanceof AbstractServerFactory) {
+      String basePath =
+          ((AbstractServerFactory) serverFactory)
+              .getJerseyRootPath()
+              .orElse(DROPWIZARD_DEFAULT_ROOT_PATH);
+
+      // fix base path by removing '/*' at the end so swagger-ui can resolve it
+      return "/*".equals(basePath) ? "/" : basePath.replaceAll("^(.*)/\\*$", "$1");
+    }
+
+    return "/api";
+  }
+
+  public interface InitialBuilder {
+
+    /**
+     * Use an existing OpenAPI 3 specification as base for the generation.
+     *
+     * <p>Note that the OpenAPI annotations always override values from the files if classes are
+     * registered with {@link #addResourcePackage(String)} or {@link
+     * #addResourcePackageClass(Class)}.
+     *
+     * <p>The postprocessing will also be done for these files, including sorting, embed parameter,
+     * and server baseurl.
+     *
+     * @param openApiJsonOrYaml the OpenAPI 3 specification as json or yaml
+     * @return the builder
+     * @see EmbedParameterModifier
+     * @see org.sdase.commons.optional.server.openapi.sort.SorterModifier
+     * @see ServerUrlFilter
+     */
+    FinalBuilder withExistingOpenAPI(String openApiJsonOrYaml);
+
+    /**
+     * Reads an existing OpenAPI 3 specification from the given classpath resource and provide it to
+     * {@link #withExistingOpenAPI(String)}
+     *
+     * @param path the path to an OpenAPI 3 YAML or JSON file in the classpath.
+     * @return the builder
+     * @see #withExistingOpenAPI(String)
+     */
+    default FinalBuilder withExistingOpenAPIFromClasspathResource(String path) {
+      return this.withExistingOpenAPI(
+          new Scanner(getClass().getResourceAsStream(path), "UTF-8").useDelimiter("\\A").next());
```

Comment:
Nice trick with `\A`.
Could you use `StandardCharsets.UTF_8` to avoid the code warning?
```suggestion
          new Scanner(getClass().getResourceAsStream(path), StandardCharsets.UTF_8).useDelimiter("\\A").next());
```

Reply:
I'm not sure what warning you refer to. But I can change that.

Reply:
Not supported yet:

![image](https://user-images.githubusercontent.com/720821/95197941-a6f0dc80-07da-11eb-9957-e670a524e299.png)

Reply:
You are right. We are still supporting Java 8. For whatever reasons, IntelliJ checked against Java 11 and produced a warning. Somehow the warning is gone now without changing the configuration.